pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -137,8 +137,12 @@ func (a *Actuator) updateMachineStatus(machine *clusterv1.Machine, awsStatus *pr
 	if networkAddresses != nil {
 		machineCopy.Status.Addresses = networkAddresses
 	}
-
-	if !equality.Semantic.DeepEqual(machine.Status, machineCopy.Status) {
+	oldAWSStatus, err := ProviderStatusFromMachine(a.codec, machine)
+	if err != nil {
+		mLog.Errorf("error updating machine status: %v", err)
+		return err
+	}
+	if !equality.Semantic.DeepEqual(awsStatus, oldAWSStatus) || !equality.Semantic.DeepEqual(machine.Status.Addresses, machineCopy.Status.Addresses) {
```

Comment:
A machine status consists of more than a list of addresses and provider specific configuration [1]. E.g. in case `.Conditions` field is changed, the status will not get updated.

[1] https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/apis/cluster/v1alpha1/machine_types.go#L81-L147

Reply:
@ingvagabund Conditions cannot be changed because see L135, after copying machine object, in machineCopy only addresses and provider status are being updated. Conditions are being copied as it is from original machine object. Therefore when comparing status of old and new machine objects, essentially only addresses and provider statuses are being checked.

Reply:
You are right we only change the network addresses and/or the provider config status at most. So right now it's ok to implement the condition this way. I would rather use exclusive comparison than inclusive. In the future we might introduce additional field we set as part of status update and easily forget to alter the condition.

Let's put explanation comment before the condition why we decided to implement it this way and add TODO note to improve it later.